core-addon/Core.js

```diff
@@ -271,6 +277,9 @@ module.exports = class Core {
         };
       }
       case "telemetry-ping": {
+        if (pausedStudies.includes(sender.id)) {
+          throw new Error(`Core._handleExternalMessage - ${sender.id} is paused and may not send telemetry`);
```

Comment:
```suggestion
          throw new Error(`Core._handleExternalMessage - ${sender.id} is paused and may not send data`);
```

Let's try not to use the term telemetry, as this might get confused with Firefox Telemetry.